inference-engine/tests/functional/plugin/shared/src/import_export_tests/import_export_base.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "import_export_tests/import_export_base.hpp"
+
+#include <fstream>
+
+namespace LayerTestsDefinitions {
+
+std::string ImportNetworkTestBase::getTestCaseName(testing::TestParamInfo<exportImportNetworkParams> obj) {
+    InferenceEngine::Precision netPrecision;
+    std::string targetDevice;
+    std::map<std::string, std::string> exportConfiguration;
+    std::map<std::string, std::string> importConfiguration;
+    std::tie(netPrecision, targetDevice, exportConfiguration, importConfiguration) = obj.param;
+
+    std::ostringstream result;
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "targetDevice=" << targetDevice << "_";
+    for (auto const& configItem : exportConfiguration) {
+        result << "_exportConfigItem=" << configItem.first << "_" << configItem.second;
+    }
+    for (auto const& configItem : importConfiguration) {
+        result << "_importConfigItem=" << configItem.first << "_" << configItem.second;
+    }
+    return result.str();
+}
+
+void ImportNetworkTestBase::Run() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    configuration.insert(exportConfiguration.begin(), exportConfiguration.end());
+    LoadNetwork();
+    Infer();
+    executableNetwork.Export("exported_model.blob");
+
+    const auto& actualOutputs = GetOutputs();
+    auto referenceOutputs = CalculateRefs();
+    Compare(referenceOutputs, actualOutputs);
+
+    for (auto const& configItem : importConfiguration) {
+        configuration[configItem.first] = configItem.second;
+    }
+    std::fstream inputStream("exported_model.blob", std::ios_base::in | std::ios_base::binary);
+    if (inputStream.fail()) {
+        FAIL() << "Cannot open file to import model: exported_model.blob";
+    }
+    auto importedNetwork = core->ImportNetwork(inputStream, targetDevice, configuration);
+    for (const auto& next_input : importedNetwork.GetInputsInfo()) {
+        ASSERT_NO_THROW(executableNetwork.GetInputsInfo()[next_input.first]);
+    }
+    for (const auto& next_output : importedNetwork.GetOutputsInfo()) {
+        ASSERT_NO_THROW(executableNetwork.GetOutputsInfo()[next_output.first]);
+    }
+    auto importedOutputs = CalculateImportedNetwork(importedNetwork);
+    Compare(importedOutputs, actualOutputs);
+}
+
+std::vector<std::vector<std::uint8_t>> ImportNetworkTestBase::CalculateImportedNetwork(InferenceEngine::ExecutableNetwork& importedNetwork) {
```

Comment:
What is the reason for such function? Should there be a component for inferring agnostic to the fact that model has been imported or loaded?

Reply:
Actually the existing code may be reused here. Fixed.